iterators_task_2/student/include/impl.h

```diff
@@ -1,21 +1,73 @@
 #pragma once
 #include <iterator>
+#include <functional>
 
 template <class Container>
 struct accumulator
 {
-    /** @todo Iterator traits*/
-    /** @todo Constructor with container*/
-    /** @todo lvalue writing operator*/
-    /** @todo rvalue writing operator*/
+    using value_type = typename Container::iterator::value_type;
+    using difference_type = typename Container::iterator::difference_type;
+    using pointer = typename Container::iterator::pointer;
+    using reference = typename Container::iterator::reference;
+    using iterator_category = std::output_iterator_tag;
 
-    /** @todo Iterator operations */
+    accumulator(Container& container)
+        : m_ref(std::ref(container))
+        , m_iter(container.end())
+    {}
+
+    accumulator(const accumulator<Container>& other) = default;
+
+    template <class AnotherContainer>
+    accumulator<Container>& operator=(AnotherContainer&& container)
+    {
+        std::copy(std::make_move_iterator(container.begin()),
+                  std::make_move_iterator(container.end()),
+                                      std::back_inserter(m_ref.get()));
+        return *this;
+    }
+
+    accumulator<Container>& operator=(value_type&& type)
+    {
+        m_ref.get().insert(m_iter, std::forward(type));
+        return *this;
+    }
+
+    accumulator& operator++()
+    {
+        ++m_iter;
```

Comment:
incrementing end iterator will result an UB, you don't need this iterator at all